reducers/plugins/index.ts

```diff
@@ -5,9 +5,15 @@ import {combineReducers} from 'redux';
 
 import remove from 'lodash/remove';
 
+import type {GenericAction} from 'mattermost-redux/types/actions';
+import {IDMappedObjects} from 'mattermost-redux/types/utilities';
+import {ClientPluginManifest} from 'mattermost-redux/types/plugins';
+
+import type {PluginComponent, ProductComponent, PostPluginComponent, AdminConsolePluginComponent} from 'types/store/plugins';
+
 import {ActionTypes} from 'utils/constants';
 
-function hasMenuId(menu, menuId) {
+function hasMenuId(menu: any, menuId: string) {
```

Comment:
Could we add an in-line type for this?
```suggestion
type Menu = {
    id: string:
    subMenu?: Menu[];
};

function hasMenuId(menu: Menu, menuId: string) {
```

Reply:
I can try, that is not exactly the valid type. Let me take a look to this.

Reply:
Fixed, this required more work that I expected, but now I think looks good :)